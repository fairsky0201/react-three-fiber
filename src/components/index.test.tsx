import React from 'react'
import TestRenderer from 'react-test-renderer'

import { Mesh, BoxBufferGeometry, MeshNormalMaterial, Line, New, Primitive } from '.'

describe('components', () => {
  test('mesh example matches snapshot', () => {
    expect(
      TestRenderer.create(
        <Mesh
          onUpdate={mesh => console.log(mesh.geometry)}
          onClick={() => console.log('click')}
          onPointerOver={() => console.log('hover')}
          onPointerOut={() => console.log('unhover')}>
          <BoxBufferGeometry attach="geometry" args={[1, 1, 1]} />
          <MeshNormalMaterial attach="material" />
        </Mesh>
      )
    ).toMatchSnapshot()
  })

  test('components.Line can be used instead of line instrinsic element', () => {
    const actual = TestRenderer.create(<Line />).toJSON()
    const expected = TestRenderer.create(<line />).toJSON()
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    expect(actual).toStrictEqual(expected!)
  })

  test('Primitive matches snapshot', () => {
    const object = { a: 10, b: 20 }
    expect(TestRenderer.create(<Primitive object={object} a={30} />)).toMatchSnapshot()
  })

  test('New matches snapshot', () => {
    class SpecialThing {
      constructor(public foo: number, _: string) {}
    }

    expect(TestRenderer.create(<New object={SpecialThing} args={[5, 'bar']} foo={2} />)).toMatchSnapshot()
  })
})
