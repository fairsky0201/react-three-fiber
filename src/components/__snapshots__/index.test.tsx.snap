// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components New matches snapshot 1`] = `
<new
  args={
    Array [
      5,
      "bar",
    ]
  }
  foo={2}
  object={[Function]}
/>
`;

exports[`components Primitive matches snapshot 1`] = `
<primitive
  a={30}
  object={
    Object {
      "a": 10,
      "b": 20,
    }
  }
/>
`;

exports[`components mesh example matches snapshot 1`] = `
<mesh
  onClick={[Function]}
  onPointerOut={[Function]}
  onPointerOver={[Function]}
  onUpdate={[Function]}
>
  <boxBufferGeometry
    args={
      Array [
        1,
        1,
        1,
      ]
    }
    attach="geometry"
  />
  <meshNormalMaterial
    attach="material"
  />
</mesh>
`;
